litestar/_openapi/schema_generation/utils.py

```diff
@@ -15,3 +26,67 @@ def sort_schemas_and_references(values: list[Schema | Reference]) -> list[Schema
         A sorted list of schemas or references
     """
     return sorted(values, key=lambda value: value.type if isinstance(value, Schema) else value.ref)  # type: ignore
+
+
+def _type_or_first_not_none_inner_type(field_definition: FieldDefinition) -> Any:
+    """Get the first inner type that is not None.
+
+    This is a narrow focussed utility to be used when we know that a field definition either represents
+    a single type, or a single type in a union with `None`, and we want the single type.
+
+    Args:
+        field_definition: A field definition instance.
+
+    Returns:
+        A field definition instance.
+    """
+    if not field_definition.is_optional:
+        return field_definition.annotation
+    inner = next((t for t in field_definition.inner_types if not t.is_none_type), None)
+    if inner is None:
+        raise ValueError("Field definition has no inner type that is not None")
+    return inner.annotation
+
+
+def _do_enum_schema(field_definition: FieldDefinition) -> bool:
```

Comment:
```suggestion
def _should_create_enum_schema(field_definition: FieldDefinition) -> bool:
```

I think that's the language we usually use for things like these. It's also consistent with the docstring :)